src/lib/Scenes/City/CitySectionList.tsx

```diff
@@ -0,0 +1,131 @@
+import { CitySectionList_city } from "__generated__/CitySectionList_city.graphql"
+import { PAGE_SIZE } from "lib/data/constants"
+import { isCloseToBottom } from "lib/utils/isCloseToBottom"
+import React from "react"
+import { createPaginationContainer, graphql, RelayPaginationProp } from "react-relay"
+import { BucketKey } from "../Map/Bucket"
+import { EventList } from "./Components/EventList"
+
+interface Props {
+  city: CitySectionList_city
+  citySlug: string
+  section: BucketKey
+  relay: RelayPaginationProp
+}
+
+class CitySectionList extends React.Component<Props> {
+  fetchData = () => {
+    const { relay } = this.props
+
+    if (relay.isLoading()) {
+      return
+    }
+    relay.loadMore(PAGE_SIZE, error => {
+      if (error) {
+        console.error("CitySectionList.tsx #fetchData", error.message)
+        // FIXME: Handle error
+      }
+    })
+  }
+
+  render() {
+    const {
+      section,
+      city: { name, shows },
+      relay,
+    } = this.props
+    return (
+      <EventList
+        key={name + section}
+        cityName={name}
+        bucket={shows.edges}
+        type={section}
+        relay={relay}
+        onScroll={isCloseToBottom(this.fetchData)}
+      />
+    )
+  }
+}
+
+export default createPaginationContainer(
+  CitySectionList,
+  {
+    city: graphql`
+      fragment CitySectionList_city on City
+        @argumentDefinitions(
+          count: { type: "Int", defaultValue: 10 }
+          cursor: { type: "String", defaultValue: "" }
+          partnerType: { type: "PartnerShowPartnerType" }
+        ) {
+        name
+        # TODO: Any way to exclude partnerType if it's not specified? When null it causes errors on MP.
```

Comment:
It's not a blocker on merging this PR, but I'd like to figure this out sometime. I couldn't find other examples of including an argument or not based on if it was present, so this might be something we need to change on MP (to not complain if `partnerType` is `null`. Thoughts @alloy?

Reply:
The error is from Gravity:

```
"https://stagingapi.artsy.net/api/v1/shows?at_a_fair=false&displayable=true&has_location=true&include_local_discovery=true&max_distance=25&near=51.51%2C-0.13&page=3&partner_types=&size=10&sort=start_at&status=current&total_count=true - {\"type\":\"param_error\",\"message\":\"Invalid parameters.\",\"detail\":{\"partner_types\":[\"does not have a valid value\"]}}"
```

So if we just don't send that from MP, I think that'd be the right choice.

Reply:
I guess you could inject a boolean variable like `hasPartnerType` and use that in a conditional directive:

```graphql
... @include(if: $hasPartnerType) {
  shows(partnerType: $partnerType, …) {
    ... CitySectionList_shows
  }
}
... @skip(if: $hasPartnerType) {
  shows(…) {
    ... CitySectionList_shows
  }
}
```

Seems a little burdensome, so perhaps a MP change would lead to cleaner code.

Reply:
I think that accepting `null` would behave more accurately according to the schema, and would avoid a crash. I've opened a PR to metaphysics here and will remove the TODO comment: https://github.com/artsy/metaphysics/pull/1560